moveit_ros/hybrid_planning/local_planner/local_planner_component/include/moveit/local_planner/trajectory_operator_interface.h

```diff
@@ -0,0 +1,107 @@
+ *     from this software without specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ *  "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ *  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ *  FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ *  COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ *  INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ *  BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ *  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ *  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ *  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ *  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ *********************************************************************/
+
+/* Author: Sebastian Jahr
+   Description: Defines an interface for a trajectory operator plugin implementation for the local planner component node.
+ */
+
+#pragma once
+
+#include <rclcpp/rclcpp.hpp>
+#include <rclcpp_action/rclcpp_action.hpp>
+
+#include <moveit_msgs/msg/robot_trajectory.hpp>
+#include <moveit_msgs/msg/constraints.hpp>
+#include <moveit/robot_state/robot_state.h>
+#include <moveit/robot_trajectory/robot_trajectory.h>
+
+namespace moveit_hybrid_planning
+{
+/**
+ * Class TrajectoryOperatorInterface - Base class for a trajectory operator. The operator's task is manage the local
+ * planner's global reference trajectory. This includes trajectory matching based on the current state to identify the
+ * current planning problem and blending of new global trajectory updates into the currently processed reference
+ * trajectory.
+ */
+class TrajectoryOperatorInterface
+{
+public:
+  /**
+   * Initialize trajectory operator
+   * @param node Node handle to access parameters
+   * @param robot_model Robot model
+   * @param group_name Name of the joint group the trajectory uses
+   * @return True if initialization was successful
+   */
+  virtual bool initialize(const rclcpp::Node::SharedPtr& node, const moveit::core::RobotModelConstPtr& robot_model,
+                          const std::string& group_name) = 0;
+
+  /**
+   * Add a new reference trajectory segment to the vector of global trajectory segments to process
+   * @param new_trajectory New reference trajectory segment to add
+   * @return True if segment was successfully added
+   */
+  virtual bool addTrajectorySegment(const robot_trajectory::RobotTrajectory& new_trajectory) = 0;
+
+  /**
+   * Return the current local constraints based on the newest robot state
+   * @param current_state Current RobotState
+   * @return Current local constraints that define the local planning goal
+   */
+  virtual robot_trajectory::RobotTrajectory getLocalTrajectory(const moveit::core::RobotState& current_state) = 0;
```

Comment:
I think this function should also be able to give the local planner feedback, as right now nothing in the `trajectory_operator_interface` can give feedback (that can be used to trigger a global replanning, for instance). Consider: 
```suggestion
  virtual moveit_msgs::action::LocalPlanner::Feedback getLocalTrajectory(const moveit::core::RobotState& current_state, robot_trajectory::RobotTrajectory& local_trajectory) = 0;
```